### PR TITLE
DAOS-10030 doc: fix Doxygen warnings and GHA technical debt.

### PR DESCRIPTION
**Pull request by kccain**
Skip-test: true
Skip-unit-tests: true

Co-authored-by: Ashley Pittman <ashley.m.pittman@intel.com>
Signed-off-by: Kenneth Cain <kenneth.c.cain@intel.com>

### CODE REVIEW

File: src/include/daos_array.h [hunk 1]
```diff
@@ -33,17 +33,15 @@ typedef struct {
...
-	 * from the largest dkey(s). This helps for checking for short reads. If
-	 * this values is not zero, then a short read is possible and should be
-	 * checked with daos_array_get_size() compared with the indexes being
-	 * read.
+	/** (on read only) number of records that are short fetched from the largest dkey(s).
+	 * Helps for checking short reads. If nonzero, a short read is possible and should be
+	 * checked with daos_array_get_size() compared with the indexes being read.
 	 */
 	daos_size_t		arr_nr_short_read;
 	daos_size_t		arr_nr_read;
```
Comment (mchaarawi) [inline src/include/daos_array.h:41]: we can add a comment on this that here:
(on read only) contains the number of records that were actually read from the array.

File: src/include/daos_cont.h [hunk 2]
```diff
@@ -12,7 +12,9 @@
 #ifndef __DAOS_CONT_H__
 #define __DAOS_CONT_H__
 
+/** container open API macro to invoke latest implementation */
 #define daos_cont_open daos_cont_open2
+/** container destroy API macro to invoke latest implementation */
 #define daos_cont_destroy daos_cont_destroy2
```
Comment (mchaarawi) [inline src/include/daos_cont.h:18]: we should probably just add something like:
Please ignore (code for backward interoperability)

**Review by mchaarawi — DISMISSED** [review]
the anchor parameters are mostly internal really, and we can probably just add something like, internal, do not use.
in any case none of my comments should keep this from approval/landing as they are additional.

**Review by mchaarawi — DISMISSED** [review]

**Comment by ashleypittman** [comment]
Looks like we could incorporate https://github.com/daos-stack/daos/pull/8314 into this PR, but we could just as well land the two seperately.

**Review by ashleypittman — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Review by johannlombardi — APPROVED** [review]

**Pull request merged**